elastic/kibana
# Preserves originating path when returning from editor

## Summary

This fixes a bug where you lose your url parameters for your workpad after editing and saving your embeddable in an editor outside of Canvas and returning to your workpad. 

In the `embeddable` plugin, I added `containerContext` which contains `getOriginatingPath` which allows presentation apps like Dashboard and Canvas to pass a function that returns the original path for the editor to return to. 

In the `maps` and `visualize` plugins, I extracted `originatingPath` from the state transfer incoming editor state and path it through to where the app returns to the originating app.

One way to test:
- Open Canvas and create a workpad
- Add any Lens/map/visualize embeddable to the workpad.
- Click on `View` > `Autoplay settings` and select `30 seconds`. Notice: the url should have the `autoplayInterval` param.
- Click on the actions menu on the embeddable to edit the embeddable
- Click `Save and return`.
- You should return to the same workpad you left, and the url should still contain the `autoplayInterval` param.


### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/master/RISK_MATRIX.mdx) |


### For maintainers
